tensorflow does not build on Darwin

### Describe the bug
```
ERROR: /private/tmp/nix-build-tensorflow-2.4.2.drv-0/output/external/com_google_protobuf/BUILD:301:11: C++ compilation of rule '@com_google_protobuf//:protoc_lib' failed (Exit 1): cc_wrapper.sh failed: error executing command external/local_config_cc/cc_wrapper.sh -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG ... (remaining 38 argument(s) skipped)
[7 / 337] checking cached actions
warning: unknown warning option '-Wno-stringop-truncation'; did you mean '-Wno-string-conversion'? [-Wunknown-warning-option]
In file included from external/com_google_protobuf/src/google/protobuf/compiler/code_generator.cc:35:
external/com_google_protobuf/src/google/protobuf/compiler/code_generator.h:41:10: fatal error: 'string' file not found
#include <string>
         ^~~~~~~~
1 warning and 1 error generated.
```

This seems to have been around for a while: https://discourse.nixos.org/t/tensorflow-on-20-09-mac/11931

### Steps To Reproduce
```nix
let
  nixpkgs = fetchTarball
    # master 2021-11-01
    "https://github.com/NixOS/nixpkgs/archive/42e9ec0c3962c4070a13ff7c403d5cde0b6237ae.tar.gz";
  pkgs = import nixpkgs { };
in with pkgs;
mkShell {
  buildInputs = [ (python3.withPackages (ps: with ps; [ tensorflow ])) ];
}
```

### Additional context
Have tried popping `NIX_CFLAGS_COMPILE = [ "-I${libcxx}/include/c++/v1" ];` locally as described [here](https://github.com/NixOS/nixpkgs/pull/26709/files#diff-fd103b21fdd23e3889978e20b10fef06cdb0551f9bf65dd8a221ecd0be744bf7R12), but it did not help.

### Notify maintainers
@jyp @abbradar 

### Metadata

 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.6.0, macOS 10.15.7`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.16`
 - channels(user): `"nixpkgs-21.05pre295466.f0869b1a2c0"
 - nixpkgs: `/Users/vg/.nix-defexpr/channels/nixpkgs`

--- a/pkgs/development/python-modules/tensorflow/default.nix
+++ b/pkgs/development/python-modules/tensorflow/default.nix
@@ -283,6 +283,7 @@ let
     bazelBuildFlags = [
       "--config=opt" # optimize using the flags set in the configure phase
     ]
+    ++ lib.optionals stdenv.cc.isClang [ "--cxxopt=-x" "--cxxopt=c++" "--host_cxxopt=-x" "--host_cxxopt=c++" ]
     ++ lib.optionals (mklSupport) [ "--config=mkl" ];
 
     bazelTarget = "//tensorflow/tools/pip_package:build_pip_package //tensorflow/tools/lib_package:libtensorflow";

